plugins/modules/zypper.py

```diff
@@ -350,9 +363,21 @@ def parse_zypper_xml(m, cmd, fail_not_found=True, packages=None):
             return parse_zypper_xml(m, cmd, fail_not_found=fail_not_found, packages=packages)
 
         return packages, rc, stdout, stderr
+
+    if m.params['simple_errors']:
+        stdout = get_simple_errors(dom)
```

Comment:
Changing this to
```suggestion
        stdout = get_simple_errors(dom) or stdout
```
would allow `get_simple_errors()` to return `None` (or an empty string) in case it doesn't find errors, and it will simply use `stdout` in that case.